spec/rspec/expectations/failure_aggregator_spec.rb

```diff
@@ -400,6 +415,18 @@ def expect_error_included_in_aggregated_failure(error)
       def fail_including
         fail { |e| expect(e.message).to include(yield) }
       end
+
+      # Each Ruby version return a different exception complement.
+      # This method gets the current version and return the
+      # right complement.
+      def exception_complement(block_levels)
+        version = RUBY_VERSION.gsub('ruby-', '') || '0'
+        if RSpec::Support::Ruby.mri?
+          version > '1.8.7' ? ":in `block (#{block_levels} levels) in <module:Expectations>'" : ''
+        else
+          version > '2.0.0' ? ":in `block in Expectations'" : ":in `Expectations'"
+        end
+      end
```

Comment:
Typically we define methods based on Ruby versions rather than perform run time changes, although these are tests I think I'd still rather have it more like this:
```suggestion
      if RSpec::Support::Ruby.mri? && RUBY_VERSION > "1.8.7"
        def exception_complement(block_levels)
          ":in `block (#{block_levels} levels) in <module:Expectations>'"
        end
      elsif RSpec::Support::Ruby.mri?
        def exception_complement(block_levels)
          ""
        end
      elsif RUBY_VERSION > "2.0.0"
        def exception_complement(block_levels)
          ":in `block in Expectations'"
        end
      else
        def exception_complement(block_levels)
          ":in `Expectations'"
        end
      end
```